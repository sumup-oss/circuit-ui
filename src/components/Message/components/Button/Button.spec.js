import React from 'react';

import { MessageButton } from '../../';

describe('MessageButton', () => {
  /**
   * Style tests.
   */
  it('should render with default styles', () => {
    const actual = create(<MessageButton />);
    expect(actual).toMatchSnapshot();
  });

  it('should render alignment styles when passed the align prop', () => {
    const actual = create(<MessageButton align={MessageButton.CENTER} />);
    expect(actual).toMatchSnapshot();
  });

  /**
   * Accessibility tests.
   */
  it('should meet accessibility guidelines', async () => {
    const wrapper = renderToHtml(<MessageButton />);
    const actual = await axe(wrapper);
    expect(actual).toHaveNoViolations();
  });
});
