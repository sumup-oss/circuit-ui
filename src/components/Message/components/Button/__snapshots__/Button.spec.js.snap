// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageButton should render alignment styles when passed the align prop 1`] = `
.circuit-0 {
  display: block;
  padding-left: 24px;
  margin-top: 16px;
  margin-left: auto;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

@media (min-width:480px) {
  .circuit-0 {
    margin-top: 0;
  }
}

<div
  class="circuit-0 circuit-1"
/>
`;

exports[`MessageButton should render with default styles 1`] = `
.circuit-0 {
  display: block;
  padding-left: 24px;
  margin-top: 16px;
  margin-left: auto;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

@media (min-width:480px) {
  .circuit-0 {
    margin-top: 0;
  }
}

<div
  class="circuit-0 circuit-1"
/>
`;
